packages/fxa-payments-server/src/routes/Subscriptions/index.tsx

```diff
@@ -62,14 +71,35 @@ export const Subscriptions = ({
   resetUpdatePayment,
   resetCancelSubscription,
   updatePaymentStatus,
+  useSCAPaymentFlow,
+  paymentUpdateStripeOverride,
+  paymentUpdateApiClientOverrides,
 }: SubscriptionsProps) => {
   const { config, locationReload, navigateToUrl } = useContext(AppContext);
 
-  const [showPaymentSuccessAlert, setShowPaymentSuccessAlert] = useState(true);
-  const clearSuccessAlert = useCallback(
-    () => setShowPaymentSuccessAlert(false),
-    [setShowPaymentSuccessAlert]
-  );
+  const [
+    updatePaymentIsSuccessViaSCA,
+    setUpdatePaymentIsSuccess,
+    resetUpdatePaymentIsSuccess,
+  ] = useBooleanState(false);
+
+  const [
+    updatePaymentSuccessAlertIsHidden,
+    hideSuccessAlert,
+    resetSuccessAlert,
+  ] = useBooleanState(false);
+
+  // Combined reset function to clear both previous success status and user's
+  // action to dismiss the alert.
+  const resetUpdatePaymentIsSuccessAndAlert = useCallback(() => {
+    resetUpdatePaymentIsSuccess();
+    resetSuccessAlert();
+  }, [resetUpdatePaymentIsSuccess]);
+
+  const showPaymentSuccessAlert =
+    !updatePaymentSuccessAlertIsHidden &&
+    // TODO: When we move to V2 SCA flow, we can drop updatePaymentStatus
```

Comment:
These bits got a little more wrinkled, but the tl;dr: is that this manages the "Your billing information has been updated successfully" alert bar for both V1 and V2 payment update paths. We can simplify this when post-SCA cleanup happens